packages/editor/src/browser/editor-manager.ts

```diff
@@ -98,6 +98,10 @@ export class EditorManager extends NavigatableWidgetOpenHandler<EditorWidget> {
     }
 
     canHandle(uri: URI, options?: WidgetOpenerOptions): number {
+        // Set the priority higher than image/pdf/etc. preview handlers if the editor is used to 'open source'
```

Comment:
i think it should return a negative number if `openFor` is not `edit`. And we should change other widget open handlers to respect `openFor` property.